src/components/dashboard/dashboard.tsx

```diff
@@ -25,42 +24,46 @@ interface IDashboardItemProps {
     claimed: boolean
     link: string
     copied: boolean
-    type: 'normal' | 'raffle'
+}
+
+interface IDashboardRaffleItemProps {
+    hash: string | undefined
+    chainId: string
+    totalAmount: string
+    token: string
+    date: string
+    claimed: boolean
+    link: string
+    copied: boolean
+    totalSlots: number
+    claimedSlots: number
+    withMFA: boolean
+    withCaptcha: boolean
+    senderName: string
+    senderAddress: string
 }
 
 export function Dashboard() {
     const [chainDetails] = useAtom(store.defaultChainDetailsAtom)
     const { address, isConnected } = useAccount()
     const router = useRouter()
-    const [localStorageData, setLocalStorageData] = useState<interfaces.ILocalStorageItem[]>([])
-    const [dashboardData, setDashboardData] = useState<IDashboardItemProps[]>([])
+    const [localStorageLinkData, setLocalStorageLinkData] = useState<interfaces.ILocalStorageItem[]>([])
+    const [localStorageRaffleData, setLocalStorageRaffleData] = useState<interfaces.ILocalStorageItem[]>([])
+    const [dashboardLinkData, setDashboardLinkData] = useState<IDashboardLinkItemProps[]>([])
+    const [dashboardRaffleData, setDashboardRaffleData] = useState<IDashboardRaffleItemProps[]>([])
     const [copiedLink, setCopiedLink] = useState<string[]>()
+    const [selectedDashboard, setSelectedDashboard] = useState<'normal' | 'raffle'>('normal')
     const gaEventTracker = hooks.useAnalyticsEventTracker('dashboard-component')
 
     const fetchLinkDetails = async (localStorageData: interfaces.ILocalStorageItem[]) => {
         try {
-            let details: IDashboardItemProps[] = []
-
+            let details: IDashboardLinkItemProps[] = []
             await Promise.all(
```

Comment:
There was an issue with dashboard loading forever. If you have multiple links, isn't promise.all a bit sketchy?

- has to wait for all to finalize
- what if one fails? is it hidden from the dashboard?

I think its better if the links all show up immediately, but the details are loading for each (instead of the entire dashboard loading)

Reply:
yes, make task about what we discussed (display emtpy/loading state & populate when getLinkDetails succeed)